src/Integration.TeamExplorer/SonarQubeNavigationItem.cs

```diff
@@ -49,6 +50,12 @@ internal SonarQubeNavigationItem([Import] ITeamExplorerController controller)
             this.IsVisible = true;
             this.IsEnabled = true;
 
+            // Note on threading:
+            // MEF constructors must be free-threaded i.e. capable of running to completion the calling thread.
+            // There's no public documentation on how these whether it's ok to create WPF artefacts like brushes
+            // and icons in the constructor. However, this is what the VS Team Explorer implementations of this
+            // class are doing, so we assume it's ok.
+            // See [VS installation directory]\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer\Microsoft.TeamFoundation.TeamExplorer.Navigation.dll
```

Comment:
It wasn't obvious whether our code was ok or not, so I found and decompiled the VS implementations to check what they did.